### PR TITLE
Replace `ShortcutKeys` `Ctrl+F` with Hotkeys

### LINKED ISSUE #9921: FileTree hotkey to search files no longer works
### Environment

- Git Extensions 4.0.0.13624
- Build c3903e122d7d97fcceb82bb66a85f2756a48558c (Dirty)
- Git 2.35.1.windows.2
- Microsoft Windows NT 10.0.22000.0
- .NET 6.0.3
- DPI 96dpi (no scaling)

### Issue description

<kbd>CTRL+F</kbd> in the FileTree panel no longer open the file search dialog:
image
image

...instead it opens the text search dialog:
image

### Steps to reproduce

Open the FileTree panel and press <kbd>CTRL+F</kbd>.

### Did this work in previous version of GitExtensions?

I haven't bisected it, but likely this is a recent regression. _Maybe_ related to https://github.com/gitextensions/gitextensions/pull/9898 or https://github.com/gitextensions/gitextensions/pull/9827?
@mstv for thoughts.

### Diagnostics

_No response_

### PR DESCRIPTION
**Pull request by mstv**
Fixes #9921

## Proposed changes

- Replace `ShortcutKeys` `Ctrl+F` with Hotkeys
- Cleanup negligences regarding Hotkey definition from #9898

### Before

grafik

### After

grafik

## Test methodology 

- manual

## Test environment(s) 

- Git Extensions 33.33.33
- Build 23e8cf2db39b6a968e8775d7e95852d12fe75a9f
- Git 2.35.1.windows.1
- Microsoft Windows NT 10.0.19044.0
- .NET 6.0.3
- DPI 96dpi (no scaling)

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Review by RussKie — APPROVED** [review]

File: GitUI/CommandsDialogs/RevisionFileTreeControl.cs [hunk 1]
```diff
@@ -257,7 +257,8 @@ public enum Command
             OpenAsTempFile = 3,
             OpenAsTempFileWith = 4,
             EditFile = 5,
-            FilterFileInGrid = 6
+            FilterFileInGrid = 6,
+            Find = 7,
```
Comment (gerhardol) [inline GitUI/CommandsDialogs/RevisionFileTreeControl.cs:261]: This Find is different from Find in RevDiff FileViewer

```suggestion
            FindFile = 7,
```

**Review by gerhardol — APPROVED** [review]
Suggest a name change
There are some assignments of ShortcutKeys still, also RevDiff and Browse...

File: GitUI/CommandsDialogs/RevisionFileTreeControl.cs (hunk 1 above)
Comment (mstv) [reply GitUI/CommandsDialogs/RevisionFileTreeControl.cs]: Renamed in both RevisionDiffControl and RevisionFileTreeControl.

Other ShortcutKeys should be dealt with separately (if necessary).

Comment (gerhardol) [reply GitUI/CommandsDialogs/RevisionFileTreeControl.cs]: In RevDiff I expected the Find to search the file contents, the searchbox exists too.
The name should be changed to something similar as in FileTree (`Find in file tree`). Just `Find file`?
I also propose that Ctrl-F by default is bound to search.

But this is likely another PR.

Comment (mstv) [reply GitUI/CommandsDialogs/RevisionFileTreeControl.cs]: > In RevDiff I expected the Find to search the file contents, the searchbox exists too. The name should be changed to something similar as in FileTree (`Find in file tree`). Just `Find file`?

Yes, already done:
> >  Renamed in _both_ RevisionDiffControl and RevisionFileTreeControl.

>  I also propose that Ctrl-F by default is bound to search.
> But this is likely another PR.

No need for a PR, just unassign the `FindFile` Hotkey.
`Ctrl+F` should have activated the file search before (via shortcut in context menu) but the Hotkey took precedence.

Comment (gerhardol) [reply GitUI/CommandsDialogs/RevisionFileTreeControl.cs]: > No need for a PR, just unassign the FindFile Hotkey.

The change is to unassign by default...

Comment (mstv) [reply GitUI/CommandsDialogs/RevisionFileTreeControl.cs]: > The change is to unassign by default...

Yes.

@RussKie, I guess you did not miss `Ctrl+F` opening the file search in the `Diff` tab. So, I expect you will agree to not assign a default Hotkey for `RevisionDiffControl.Command.FindFile`, don't you?

**Review by gerhardol — APPROVED** [review]

**Review by RussKie — COMMENTED** [review]
👍 

We just need to add auto-focus for the File Tree control:
```diff
diff --git a/GitUI/CommandsDialogs/FormBrowse.cs b/GitUI/CommandsDialogs/FormBrowse.cs
index 9f10dddd3..726e10e26 100644
--- a/GitUI/CommandsDialogs/FormBrowse.cs
+++ b/GitUI/CommandsDialogs/FormBrowse.cs
@@ -1549,6 +1549,10 @@ private void CommitInfoTabControl_SelectedIndexChanged(object sender, EventArgs
                 // workaround to avoid focusing the "filter files" combobox
                 revisionDiff.SwitchFocus(alreadyContainedFocus: false);
             }
+            else if (CommitInfoTabControl.SelectedTab == TreeTabPage)
+            {
+                fileTree.SwitchFocus(alreadyContainedFocus: false);
+            }
         }
```

File: GitUI/CommandsDialogs/RevisionFileTreeControl.cs (hunk 1 above)
Comment (RussKie) [reply GitUI/CommandsDialogs/RevisionFileTreeControl.cs]: Yes, that's fine.

**Comment by mstv** [comment]
> We just need to add auto-focus for the File Tree control

👍

Ready for squash-merge.

**Comment by RussKie** [comment]
Go for it :)

**Pull request merged**